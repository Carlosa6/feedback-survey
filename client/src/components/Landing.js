import React from 'react';
import { connect } from 'react-redux';

export const Landing = ({ loggedInUser }) => {

    return (
        <div style={{ textAlign: 'center' }}>
            <h1>Emaily</h1>
            <h5>{loggedInUser ? 'User is logged in' : 'Please login to continue'}</h5>
            <p>Come here to collect Feedback!</p>
        </div>
    );
}

const mapStateToProps = (state) => {
    return {
        loggedInUser: state.auth
    }
}

export default connect(mapStateToProps, null)(Landing);