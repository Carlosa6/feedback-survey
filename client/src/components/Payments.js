import React, { Component } from 'react';
import StripeCheckout from 'react-stripe-checkout';
import { connect } from 'react-redux';
import * as actions from '../actions/index';

export class Payments extends Component {
    render() {
        return (
            <StripeCheckout
                name="Emaily"
                description="$50 for 50 email credits"
                amount={5000}
                token={token => this.props.handleToken(token)}
                stripeKey={process.env.REACT_APP_STRIPE_KEY ? process.env.REACT_APP_STRIPE_KEY : 'test_key'}>
                <button className="btn-flat">
                    Add Credits
                </button>
            </StripeCheckout>
        )
    }
}


export default connect(null, actions)(Payments);