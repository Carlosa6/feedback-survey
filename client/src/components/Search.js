import React, { Component } from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field } from 'redux-form';
import SurveyField from './Surveys/SurveyField';
import * as actions from '../actions/index';


class Search extends Component {
    state = { searchValue: "" };
    constructor(props) {
        super(props);
        this.state = { searchValue: "" };
    }

    renderContent(surveys) {
        return surveys.map((survey, index) => {
            return (
                <div key={index}>
                    <div key={survey['title']}>
                        <label>{survey['title']}</label>
                    </div>
                    <div key={index + `body`}>
                        {survey['body']}
                    </div>
                </div>
            );
        });
    }
    render() {
        return (
            <div className="container ">
                <div className="space-space-top1 parallax">
                    <form >
                        <Field component={SurveyField} type="text" label="SEARCH" name="search" key="search" />
                        <button className="teal btn-flat center white-text" onClick={(evt) => { this.props.searchSurvey(this.props.searchValue.values['search']); evt.preventDefault(); }}>Click me!</button>
                    </form>
                </div>
                {this.props.searchData ? this.renderContent(this.props.searchData) : null}
            </div >
        );
    }
};

function mapStateToProps({ form, surveys }) {
    debugger;
    return {
        searchValue: form.searchForm,
        searchData: surveys
    }
}


export default reduxForm({
    form: 'searchForm'
})(connect(mapStateToProps, actions)(Search));