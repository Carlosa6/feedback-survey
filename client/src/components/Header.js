import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import Payments from './Payments';

export class Header extends Component {


    renderContent() {
        switch (this.props.auth) {
            case null: return;
            case false: return (
                <li><a href="/auth/google">Login with google</a></li>
            );
            default: return [
                <li key="surveys"><Link to={this.props.auth ? '/surveys/new' : '/'}><button className="btn-flat" >
                    Create Surveys </button></Link></li>,
                <li key="payment"><Payments /></li>,
                <li key="credits" style={{ margin: '0 10px' }}>Credits:{this.props.auth.credits}</li>,
                <li key="logout"> <a href="/api/logout">Logout</a> </li>];
        }
    }

    routehandler() {
        this.props.history.push('/surveys/new');
    }

    render() {
        return (
            <nav>
                <div className="nav-wrapper">
                    <Link
                        to={this.props.auth ? '/surveys' : '/'}
                        className="brand-logo">Emaily</Link>
                    <ul id="nav-mobile" className="right hide-on-med-and-down">
                        {this.renderContent()}
                    </ul>
                </div>
            </nav >
        );
    }
}

function mapStatesToProp({ auth }) {
    return { auth }
}

export default connect(mapStatesToProp, null)(Header);