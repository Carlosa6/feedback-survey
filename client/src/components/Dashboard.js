import React from 'react';
import { Link } from 'react-router-dom';
import Search from './Search';

const Dashboard = () => {
    return (
        <div style={{ textAlign: 'center' }}>
            <h1>Dashboard</h1>
            <div><Search /></div>
            <div className="fixed-action-btn space-top1">
                <Link
                    style={{ position: 'absolute', bottom: '50px', right: '50px' }} to="/surveys/new" href="#" className="btn-floating btn-large red">
                    <i className="large material-icons">add</i>
                </Link>
            </div>
        </div >
    );
}

export default Dashboard;