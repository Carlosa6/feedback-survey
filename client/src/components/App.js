import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import Header from './Header';
import Landing from './Landing';
import Dashboard from './Dashboard';
import SurveyNew from './Surveys/SurveyNew';

class App extends Component {

    componentDidMount() {
        this.props.fetchUser();
    }

    render() {
        return (
            <div className="container" >
                <div className="container">
                    <Header />
                    <Route exact path="/" component={Landing} />
                    <Route exact path="/surveys" component={Dashboard} />
                    <Route path="/surveys/new" component={SurveyNew} />
                </div>
            </div >
        );
    }
};

export default withRouter(connect(null, actions)(App));