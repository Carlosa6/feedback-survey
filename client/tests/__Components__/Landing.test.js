import React from 'react';
import Enzyme, { shallow, mount, render } from "enzyme";
import { shallowToJson } from 'enzyme-to-json';
import { Landing } from '../../src/components/Landing';
import Adapter from 'enzyme-adapter-react-16';
const loggedIn = true;

Enzyme.configure({ adapter: new Adapter() });
let wrapper;
describe("Landing Component", () => {
    beforeAll(() => {
        wrapper = mount(<Landing />);
    });
    test('renders Landing component correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
