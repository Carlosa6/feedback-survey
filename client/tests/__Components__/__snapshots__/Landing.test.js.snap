// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component renders Landing component correctly 1`] = `
<Component>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h1>
      Emaily
    </h1>
    <h5>
      Please login to continue
    </h5>
    <p>
      Come here to collect Feedback!
    </p>
  </div>
</Component>
`;
